frame/democracy/src/lib.rs

```diff
@@ -1677,19 +1818,102 @@ impl<T: Trait> Module<T> {
 	}
 
 	/// Current era is ending; we should finish up any proposals.
-	fn begin_block(now: T::BlockNumber) -> DispatchResult {
+	///
+	/// If a referendum is launched or maturing take full block weight.
+	fn begin_block(now: T::BlockNumber) -> Result<Weight, DispatchError> {
+		// TODO TODO: do idle weight using benchmark on_initialize_no_launch_no_maturing
+		let mut weight = 0;
 		// pick out another public referendum if it's time.
 		if (now % T::LaunchPeriod::get()).is_zero() {
 			// Errors come from the queue being empty. we don't really care about that, and even if
 			// we did, there is nothing we can do here.
 			let _ = Self::launch_next(now);
+			weight = T::MaximumBlockWeight::get();
 		}
 
 		// tally up votes for any expiring referenda.
 		for (index, info) in Self::maturing_referenda_at(now).into_iter() {
 			let approved = Self::bake_referendum(now, index, info)?;
 			ReferendumInfoOf::<T>::insert(index, ReferendumInfo::Finished { end: now, approved });
+			weight = T::MaximumBlockWeight::get();
+		}
+
+		Ok(weight)
+	}
+
+	/// Reads the length of PublicProps without getting the complete value in the runtime.
+	fn len_of_public_props() -> u32 {
```

Comment:
Just a note, would be so cool to get a generalized version of this

Reply:
we already have `PublicProps::<T>::decode_len()`.

Reply:
Which is great but only works for `Vec`s (and `Vec`-likes?)

Reply:
But yes, in this case `decode_len` could be used.

Reply:
I guess I was conflating collection length with byte length in the original comment.

Reply:
contrary to decode_len this function doesn't get the raw value in the runtime but yes this might be not that needed. I can just use decode_len for this one.

Reply:
fixed throught https://github.com/paritytech/substrate/pull/5975